### DISCUSSION TITLE
xframe custom setting ignored, defaults to same origin

### DISCUSSION THREAD

**Discussion by hkovacs** [strapi/strapi#6762] [Legacy Topics]
**Describe the bug**
documents served by my api.dev.mysite.com on dev.mysite.com are throwing `Refused to display 'https://api.dev.mysite.com/uploads/my-file.pdf' in a frame because it set 'X-Frame-Options' to 'sameorigin'.`

this was working fine before the upgrade to v3 stable with custom setting in `/path/to/strapi/config/env/development/security.json`:
```
  "xframe": {
    "enabled": true,
    "value": "ALLOW-FROM dev.mysite.com"
  },
```

**Steps to reproduce the behavior**
see above

**Expected behavior**
since this was working pre stable v3. i expect that it should still work in stable v3.0.4

**Screenshots**
none

**Code snippets**
see above

**System**
Node.js version: v10.16.3
NPM version: 6.9.0
Strapi version: Strapi v3.0.4
Database: mongo v4.0.12
Operating system: ubuntu 18.04

**Additional context**
Thank you!

**Comment by derrickmehaffy** [comment]
`Strapi v3.0.0-beta.20.3` Strapi beta versions are no longer supported, the only instance where it is would be during a migration to the stable release.

I am marking this issue as closed but if you can reproduce the issues on the stable release please do reply back and the issue can be reviewed again.

**Comment by hkovacs** [comment]
sorry, i pasted that in and forgot to modify. this is broken on stable versions v3.0.1 through v3.0.4

**Comment by derrickmehaffy** [comment]
FYI there is no more `security.json` in the stable release

**Comment by derrickmehaffy** [comment]
Security middleware: https://strapi.io/documentation/v3.x/concepts/middlewares.html#security-middlewares

Configure all middleware: https://strapi.io/documentation/v3.x/concepts/middlewares.html#configuration-and-activation

**Comment by hkovacs** [comment]
ok, well that explains it (head smack)

whats not clear to me is how to create `.env` file for each environment `development, staging, production` like existed in 3-beta.

things made sense with `config/env/development`. are we supposed to use `config/.env.development` now? this is not mentioned in the docs.

thank you!

**Comment by hkovacs** [comment]
i guess my pressing issue here is, how to implement `middleware.js` `xframe` custom value for development and production?

**Comment by derrickmehaffy** [comment]
@hkovacs you are on the Strapi slack right?

**Reply by derrickmehaffy** [reply]
BTW moving this to a discussion as it's not a bug

**Comment by hkovacs** [comment]
my solution to this is as follows:

firstly, make file in `config/env/{myenvironment}/middleware.js`
next set this code for starters:
```
module.exports = ({ env }) => ({
    //...
    timeout: 100,
    load: {
        before: ['responseTime', 'logger', 'cors', 'responses', 'gzip'],
        order: [
            "Define the middlewares' load order by putting their name in this array in the right order",
        ],
        after: ['parser', 'router'],
    },
    settings: {
        cors: {
            enabled: true,
            origin: 'https://dev.mysite.com, https://api.dev.mysite.com'
        },
        xframe: {
            enabled: true,
            value: 'ALLOW-FROM dev.mysite.com'
        },
    },
});

```

important note to be aware that x-frame `ALLOW-FROM` is obsolete and really not secure in strapi at this time. see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options

**Reply by AlexGrafe** [reply]
Thanks. This solved it for me as well.
